spec/features/links_spec.rb

```diff
@@ -1,91 +1,50 @@
 describe 'links' do
-  before do
-    visit '/'
-  end
+  before { visit '/' }
 
-  context 'header/login/logout' do
-    before do
-      find('#link-to-login').click
+  after { logout }
 
-      fill_in 'spree_user_email', with: ENV['USERNAME_SPREE']
-      fill_in 'spree_user_password', with: ENV['PASSWORD_SPREE']
+  context 'Header' do
+    it 'Redirects to My Account' do
+      login(ENV['USERNAME_SPREE'], ENV['PASSWORD_SPREE'])
+      click_link 'My Account'
 
-      find('.btn.btn-lg').click
+      expect(page).to have_css('h1', text: 'My Account')
     end
 
-    after do
-      visit '/logout'
-    end
+    it 'Check logo' do
+      find('#logo').click
 
-    it 'user can login' do
-      expect(page).to have_css('.navbar-right', text: 'My Account')
-    end
-
-    it 'click my account' do
-      find('#nav-bar [href="/account"]').click
-      expect(page).to have_css('h1', text: /My Account/i)
-    end
-  end
-
-  context 'header/no-login' do
-    it 'check spree' do
-      find('#logo [href="/"]').click
       expect(page.current_url).to eq "#{Capybara.app_host}/"
     end
 
-    it 'check home' do
-      find('.info', text: 'Ruby on Rails Tote').click
+    it 'Check home' do
+      first('.product-list-item').click
       find('#home-link [href="/"]').click
+
       expect(page).to have_link(nil, href: '/')
     end
 
-    it 'check cart' do
+    it 'Check cart' do
       find('#link-to-cart [href="/cart"]').click
```

Comment:
Href unnecessary

Reply:
It now magically works without providing additional href information, I wonder why coz it was not working with the ID only before <thinking_face>